modules/unsupported/mbtiles/src/main/java/org/geotools/mbtiles/MBTilesMetadata.java

```diff
@@ -1,17 +1,48 @@
 package org.geotools.mbtiles;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.referencing.CRS;
+import org.geotools.util.logging.Logging;
 import org.opengis.geometry.Envelope;
+import org.opengis.geometry.MismatchedDimensionException;
+import org.opengis.referencing.FactoryException;
 
 public class MBTilesMetadata {
     
+    private static Logger LOGGER = Logging.getLogger(MBTilesMetadata.class); 
+    
     protected static Pattern patternEnvelope = Pattern.compile(" *(\\-?[0-9\\.]*) *, *(\\-?[0-9\\.]*) *, *(\\-?[0-9\\.]*) *, *(\\-?[0-9\\.]*) *");
     
-    public enum t_type {OVERLAY, BASE_LAYER} ;
+    public enum t_type {
+        OVERLAY("overlay"), 
+        BASE_LAYER("baselayer");
+        
+        private t_type(String identifier) {
+            this.identifier = identifier;
+        }
+        
+        public final String identifier;
+        
+        public static t_type lookUp(final String s) {
+            return Arrays.stream(t_type.values())
+                .filter((t)->t.identifier.equalsIgnoreCase(s))
+                .findAny()
+                .orElseThrow(()->new IllegalArgumentException(String.format("Unknown mbtiles type '%s'",s)));
```

Comment:
From an API standpoint, shouldn't the lookup method have `throws IllegalArgumentException`?
I am not quite sure how this works with the Java 8 constructs you are using here?